File: leetcode/128/128.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int, int> existed;
        for(int i = 0; i < nums.size(); i++) {
            existed[nums[i]] = 1;
        }
        int maxLen = 0, currentLen, currentNum;
        for(int i = 0; i < nums.size(); i++) {
            currentNum = nums[i];
            currentLen = 0;
            if(!existed[currentNum - 1]) {
                while(existed[currentNum]) {
                    currentNum++;
                    currentLen++;
                }
            }
            maxLen = max(maxLen, currentLen);
        }
        return maxLen;
    }
};

File: leetcode/128/128.java
class Solution {
    public int longestconsecutive(int[] nums) {
        Set<Integer> num_set = new HashSet<Integer>();

        for(int num : nums) {
            num_set.add(num);
        }

        int longestSreak = 0;

        for(int num : nums) {
            if(!num_set.contains(num - 1)) {
                int currentNum = num;
                int currentStreak = 1;

                while(num_set.contains(currentNum + 1)) {
                    currentNum += 1;
                    currentStreak += 1;
                }

                longestSreak = Math.max(currentStreak, longestSreak);
            }
        }

        return longestSreak;
    }
}